internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2651,3 +2704,66 @@ func flattenKubernetesClusterMaintenanceConfigurationTimeInWeeks(input *[]contai
 	}
 	return results
 }
+
+func expandKubernetesClusterHttpProxyConfig(input []interface{}) *containerservice.ManagedClusterHTTPProxyConfig {
+	httpProxyConfig := containerservice.ManagedClusterHTTPProxyConfig{}
+	if len(input) == 0 {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
+	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
+	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))
+	noProxyRaw := config["no_proxy"].([]interface{})
+	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)
+	return &httpProxyConfig
+}
+
+func flattenKubernetesClusterHttpProxyConfig(props *containerservice.ManagedClusterProperties, flattenedDefaultNodePool *[]interface{}) []interface{} {
+	results := make(map[string]interface{})
+	httpProxyConfig := props.HTTPProxyConfig
+	if httpProxyConfig != nil {
+		results["http_proxy"] = httpProxyConfig.HTTPProxy
+		results["https_proxy"] = httpProxyConfig.HTTPSProxy
```

Comment:
both of these are pointers so would need to be nil-checked given how the Azure API behaves for older resources:

```suggestion
		httpProxy := ""
		if httpProxyConfig.HTTPProxy != nil {
			httpProxy = *httpProxyConfig.HTTPProxy
		}
		results["http_proxy"] = httpProxy
		
		httpsProxy := ""
		if httpProxyConfig.HTTPSProxy != nil {
			httpsProxy = *httpProxyConfig.HTTPSProxy
		}
		results[""] = httpsProxy
```